src/dag-node-factory.js

```diff
@@ -0,0 +1,193 @@
+'use strict'
+
+const stable = require('stable')
+const sort = stable.inplace
+// const mh = require('multihashes')
+const protobuf = require('protocol-buffers')
+const proto = protobuf(require('./dag.proto'))
+const util = require('./util')
+const DAGNode = require('./dag-node')
+const DAGLink = require('./dag-link')
+
+class DAGNodeFactory {
+  static create (data, dagLinks, hashAlg, callback) {
```

Comment:
wrap the create in `promisify`, so that it supports both.
plus add a test for promises